### PR TITLE
fix(modal): allow reopening after close

### PR DESCRIPTION
**Pull request by thallada**
Modal should re-open if a parent updates its props to `{open: true}`.

### CODE REVIEW

File: src/Modal/index.jsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ class Modal extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.open !== this.props.open) {
+    if (nextProps.open !== this.state.open) {
```
Comment (jaebradley) [inline src/Modal/index.jsx]: Apologies in advance for the rubber 🦆 -ing.

Will this cause a fundamentally different behavior in `Modal`?

Consider this newly-initialized, simplified `Modal`

```javascript
<Modal open />
```

The `open` `state` value is `true` and the `open` `prop` value is `true`.

Let's say I close this `Modal`.

For reference, here's the `close` method

```javascript
close() {
  this.setState({ open: false });
  this.props.onClose();
}
```

So now, my `Modal`'s `open` `state` value is `false` and the `open` `prop` value is `true`.

`componentWillReceiveProps` has not been called yet because, well, the `Modal` hasn't received any new `props`.

Let's say we update the `Modal` dynamically so that it's `prop` value is now

```javascript
<Modal open={false} />
```

So in the old implementation, `nextProps.open` (which is `false`) is not equal to `this.props.open` (which is `true`), so then `setState` would get called, and a new `render` would occur, but the `open` `state` value would remain `false`.

In the new implementation, `nextProps.open` is equal to `this.state.open` (which is `false`) so the `setState` will not get called.

In both cases, the `Modal` will remain closed.

**tl;dr: It doesn't look like this will change the behavior post-close, but it might eliminate a `render`**

**bonus tl;dr: If you got here, thanks for being patient**

Comment (jaebradley) [reply src/Modal/index.jsx]: Style nit - consider destructuring for added tersnessosity (not a word, but should be for irony)

```javascript
componentWillReceiveProps({ open }) {
  if (open !== this.state.open) {
    this.setState({ open });
  }
}
```

Comment (thallada) [reply src/Modal/index.jsx]: Thanks for thoroughly checking the post-close behavior!
